extensions/blocks/story/edit.native.js

```diff
@@ -0,0 +1,301 @@
+import { __, sprintf } from '@wordpress/i18n';
+import { getProtocol } from '@wordpress/url';
+import { doAction, hasAction } from '@wordpress/hooks';
+import {
+	// TODO implement the similar / following bridge signals
+	// requestImageFailedRetryDialog,
+	// requestImageUploadCancelDialog,
+	// requestImageFullscreenPreview,
+	mediaUploadSync,
+	mediaSaveSync,
+	requestMediaFilesEditorLoad,
+} from '@wordpress/react-native-bridge';
+
+/**
+ * Internal dependencies
+ */
+import styles from './editor.scss';
+import StoryEditingButton from './story-editing-button';
+
+class StoryEdit extends React.Component {
+	constructor( props ) {
+		super( props );
+
+		this.onEditButtonTapped = this.onEditButtonTapped.bind( this );
+
+		this.mediaUploadStateReset = this.mediaUploadStateReset.bind( this );
+		this.finishMediaUploadWithSuccess = this.finishMediaUploadWithSuccess.bind( this );
+		this.finishMediaUploadWithFailure = this.finishMediaUploadWithFailure.bind( this );
+		this.updateMediaUploadProgress = this.updateMediaUploadProgress.bind( this );
+
+		// save progress bindings
+		this.mediaSaveStateReset = this.mediaSaveStateReset.bind( this );
+		this.finishMediaSaveWithSuccess = this.finishMediaSaveWithSuccess.bind( this );
+		this.finishMediaSaveWithFailure = this.finishMediaSaveWithFailure.bind( this );
+		this.updateMediaSaveProgress = this.updateMediaSaveProgress.bind( this );
+
+		this.onStorySaveResult = this.onStorySaveResult.bind( this );
+
+		this.onMediaModelCreated = this.onMediaModelCreated.bind( this );
+
+		this.state = {
+			isUploadInProgress: false,
+			isSaveInProgress: false,
+			didUploadFail: false,
+			didSaveFail: false,
+		};
+	}
+
+	componentDidMount() {
+		const { attributes } = this.props;
+		if ( attributes.mediaFiles !== undefined ) {
+			for ( let i = 0; i < attributes.mediaFiles.length; i++ ) {
+				const protocolForUrl = getProtocol( attributes.mediaFiles[ i ].url );
+				if (
+					( attributes.mediaFiles[ i ].id && protocolForUrl !== 'http:' ) ||
+					protocolForUrl !== 'https:'
+				) {
+					mediaUploadSync();
+					mediaSaveSync();
+					return;
+				}
+			}
+		}
+	}
+
+	componentWillUnmount() {
+		// this action will only exist if the user pressed the trash button on the block holder
+		// TODO check whether blocks.onRemoveBlockCheckUpload really is being set elsewhere
+		// and ditch / fix if not
+		if (
+			hasAction( 'blocks.onRemoveBlockCheckUpload' ) &&
+			( this.state.isUploadInProgress || this.state.isSaveInProgress )
+		) {
+			const { attributes } = this.props;
+			if ( attributes.mediaFiles !== undefined ) {
+				for ( let i = 0; i < attributes.mediaFiles.length; i++ ) {
+					const protocolForUrl = getProtocol( attributes.mediaFiles[ i ].url );
+					if (
+						( attributes.mediaFiles[ i ].id && protocolForUrl !== 'http:' ) ||
+						protocolForUrl !== 'https:'
+					) {
+						doAction( 'blocks.onRemoveBlockCheckUpload', attributes.mediaFiles[ i ].id );
+					}
+				}
+			}
```

Comment:
If this logic is needed here and above in `componentDidMount`, it could be moved into a helper.

Reply:
👍 moved in de0d8d5be5 as well